evalml/automl/engine/sequential_engine.py

```diff
@@ -1,105 +1,57 @@
-import sys
-import traceback
-
-import numpy as np
-
-from evalml.automl.engine import EngineBase
-from evalml.exceptions import PipelineScoreError
-from evalml.model_family import ModelFamily
+from evalml.automl.engine.engine_base import (
+    EngineBase,
+    EngineComputation,
+    evaluate_pipeline,
+    train_pipeline
+)
 from evalml.objectives.utils import get_objective
-from evalml.utils import get_logger
-
-logger = get_logger(__file__)
 
 
-class SequentialEngine(EngineBase):
-    """The default engine for the AutoML search. Trains and scores pipelines locally, one after another."""
-
-    def evaluate_batch(self, pipelines):
-        """Evaluate a batch of pipelines using the current dataset and AutoML state.
+class SequentialComputation(EngineComputation):
+    """A Future-like api for jobs created by the SequentialEngine."""
 
-        Arguments:
-            pipelines (list(PipelineBase)): A batch of pipelines to be fitted and evaluated.
+    def __init__(self, work, **kwargs):
+        self.work = work
+        self.kwargs = kwargs
+        self.meta_data = {}
 
-        Returns:
-            list (int): a list of the new pipeline IDs which were created by the AutoML search.
-        """
-        if self.X_train is None or self.y_train is None:
-            raise ValueError("Dataset has not been loaded into the engine.")
-        new_pipeline_ids = []
-        index = 0
-        while self._should_continue_callback() and index < len(pipelines):
-            pipeline = pipelines[index]
-            self._pre_evaluation_callback(pipeline)
-            X, y = self.X_train, self.y_train
-            if pipeline.model_family == ModelFamily.ENSEMBLE:
-                X, y = self.X_train.iloc[self.ensembling_indices], self.y_train.iloc[self.ensembling_indices]
-            elif self.ensembling_indices is not None:
-                training_indices = [i for i in range(len(self.X_train)) if i not in self.ensembling_indices]
-                X = self.X_train.iloc[training_indices]
-                y = self.y_train.iloc[training_indices]
-            evaluation_result = EngineBase.train_and_score_pipeline(pipeline, self.automl, X, y)
-            new_pipeline_ids.append(self._post_evaluation_callback(pipeline, evaluation_result))
-            index += 1
-        return new_pipeline_ids
+    def done(self) -> bool:
+        """Is the computation done?"""
+        return True
```

Comment:
Why do we auto-return just true here?

Reply:
I should have added some comments, this is confusing 🙈 

The issue is that in order to separate the engine from the AutoMLSearch loop, we need the sequential computations to behave the same way as concurrent computations from AutoMLSearch's point-of-view. One way to do this is by delaying the computation in the sequential engine until `get_result` is called. Since AutoMLSearch will call `get_result` only when the computation is "done", by always returning `True` we make sure that `get_result` is called in the order that the jobs are submitted. So the computations happen sequentially!